packages/rx_bloc_cli/mason_templates/bricks/feature_onboarding/__brick__/lib/feature_onboarding/blocs/onboarding_phone_bloc.dart

```diff
@@ -0,0 +1,134 @@
+  Stream<CountryCodeModel?> get countryCode;
+
+  /// The phone number
+  Stream<String> get phoneNumber;
+
+  /// The state of the submit phone number button
+  Stream<bool> get submitPhoneNumberEnabled;
+
+  /// Error state of the bloc
+  Stream<ErrorModel> get errors;
+
+  /// Error state as a result of a validation
+  @RxBlocIgnoreState()
+  Stream<ErrorModel?> get validationErrors;
+}
+
+/// The [OnboardingPhoneBloc] handles the user's phone number setup
+@RxBloc()
+class OnboardingPhoneBloc extends $OnboardingPhoneBloc {
+  OnboardingPhoneBloc(this._userService) {
+    errorWithTagState
+        .map((errorWithTag) => errorWithTag.tag == tagValidation
+            ? errorWithTag.exception.asErrorModel()
+            : null)
+        .bind(_validationErrorSubject);
+  }
+
+  /// The user service used to communicate the user phone number
+  final UserService _userService;
+
+  /// Tag used to denote the submit phone number event
+  static const tagSubmitPhoneNumber = 'SubmitPhoneNumber';
+
+  /// Tag used to denote validation results
+  static const tagValidation = 'Validation';
+
+  /// Subject to handle validation errors
+  final _validationErrorSubject = BehaviorSubject<ErrorModel?>();
+
+  @override
+  Stream<ErrorModel> _mapToErrorsState() => errorWithTagState
+      .where((errorWithTag) => errorWithTag.tag == tagSubmitPhoneNumber)
+      .map((error) => error.exception)
+      .mapToErrorModel();
+
+  @override
+  Stream<ErrorModel?> get validationErrors => _validationErrorSubject;
+
+  @override
+  Stream<bool> _mapToIsLoadingState() => loadingWithTagState
+      .where((loadingWithTag) => loadingWithTag.tag == tagSubmitPhoneNumber)
+      .map((loadingWithTag) => loadingWithTag.loading)
+      .distinct();
+
+  @override
+  Stream<CountryCodeModel?> _mapToCountryCodeState() => _$setCountryCodeEvent;
+
+  @override
+  Stream<String> _mapToPhoneNumberState() => _$setPhoneNumberEvent
+      .map((phoneNumber) => phoneNumber.replaceAll(' ', ''))
+      .startWith('')
+      .distinct();
+
+  @override
+  Stream<UserModel> _mapToUserState() => _$submitPhoneNumberEvent
+      .debounceTime(const Duration(milliseconds: 500))
+      .withLatestFrom2(
+          countryCode,
+          phoneNumber,
+          (_, CountryCodeModel? countryCode, String phoneNumber) =>
+              '+${countryCode!.code}$phoneNumber')
+      .switchMap((fullPhoneNumber) => _userService
+          .submitPhoneNumber(fullPhoneNumber)
+          .asResultStream(tag: tagSubmitPhoneNumber))
+      .setResultStateHandler(this)
+      .whereSuccess()
+      .asBroadcastStream();
+
+  @override
+  Stream<bool> _mapToSubmitPhoneNumberEnabledState() => Rx.merge<bool>([
+        Rx.combineLatest2(
+          countryCode.startWith(null),
+          phoneNumber.skip(1), // skip the initial empty value
+          (CountryCodeModel? countryCode, String phoneNumber) =>
+              _validatePhoneNumber(countryCode?.code, phoneNumber),
```

Comment:
Validations must be done in the domain business logic. a.k.a into a service

Reply:
Fixed